### ISSUE TITLE
Increasing input resolution for visual token pruning & Fine-tuning suggestions

### ISSUE THREAD

**Issue by Zeyuxia** [declare-lab/nora#27]
Hi NORA team,

Thanks for the amazing work and for open-sourcing the codebase!

I am currently researching Visual Token Pruning algorithms for VLA models. To thoroughly stress-test my pruning algorithm with a higher visual token load and multi-view inputs, I need to increase NORA's default input resolution (e.g., from 224x224 to 448x448 or 672x672).

Since directly increasing the resolution during inference will likely cause a drop in the task success rate (due to out-of-distribution spatial embeddings), I plan to fine-tune the model on a high-resolution subset of the dataset to help the model adapt and maintain its accuracy.

Could you please provide some guidance on the following two points?

1. Codebase modifications for higher resolution:
Besides changing the image transforms/resize parameters in the dataset loader, are there any hardcoded constraints I should be aware of? Given that the backbone is Qwen2.5-VL (which supports dynamic resolution), do I need to manually adjust any max_pixels settings or positional encoding parameters in the forward pass to prevent the model from down-sampling the images back?

2. Fine-tuning strategy for High-Res adaptation:
To recover the action success rate at a higher resolution, I plan to use LoRA for a lightweight fine-tuning phase. Which modules would you highly recommend targeting? Would it be sufficient to only fine-tune the Vision-Language Projector and the final Action Prediction layers, or is it necessary to apply LoRA to the LLM backbone as well?

Any advice or pointers to the specific code files would be greatly appreciated. Thank you!

**Comment by hungchiayu1** [comment]
Hi, if you notice in the message template, the image is being resized to 224x224, you may wish to modify this if you want to increast the input resolution. Furthermore, if you are using RLDS dataset, I believe it defaults to resizing to 224 and you need to change that in the dataloader. If you are using Lerobot dataset, ensure your dataset is loaded at the resolution you want and you can resize through Qwen2.5-VL. 

A thing to note that is that you have to ensure the resizing strategy you used for training is consistent with inference
https://github.com/declare-lab/nora/blob/6b18c23d7875052e03fba4f8c2f32bd6a8a5c4a9/lerobot_training/lerobot_training.py#L135-L145

**Comment by Zeyuxia** [comment]
Thank you very much! and we will cite your work.
